Add branch parameter to generate intended config view (#871)
nautobot/nautobot-app-golden-config: docs/user/app_feature_intended.md

The returned response will contain the rendered configuration for the specified device, the GraphQL data that was used, and if applicable, a diff of the most recent intended config that was generated by the **Intended Configuration** job. The web UI provides a simple form to interact with this REST API. You can access the web UI by clicking on "Generate Intended Config" in the "Tools" section of the Golden Config navigation menu.

> For more advanced use cases, the REST API and web UI also accept an optional `graphql_query_id` parameter to specify a custom GraphQL query to use when rendering the configuration. If a `graphql_query_id` is not provided, the default query configured in the Device's Golden Config settings will be used.

Reviewer: For this whole docs section (Developing Intended Configuration Templates), could you please make a small change and flip the order? You start talking about the feature and how to use it via the API, but given we're in the "User Guide" let's start with the screenshot of the UI and what they can do there (what inputs they can provide, the 3 panels, what's shown there, noting the git pull as well etc.)

And then at the end provide the option of using the REST API endpoints (I'd even have that info under developer guide, but I don't see any of that in GC's docs so nevermind).

Author: I made these changes

![image](https://github.com/user-attachments/assets/f2ae8a92-fa91-4cb0-8c3e-c80f42147e8d)

Author: Updated

![image](https://github.com/user-attachments/assets/7ae59321-76e3-42a7-b4d9-c3de656120f3)

Reviewer: Looks great, thank you!

> For more advanced use cases, the REST API and web UI also accept a `graphql_query_id` parameter to specify a custom GraphQL query to use when rendering the configuration. If a `graphql_query_id` is not provided, the default query configured in the Device's Golden Config settings will be used.

Author: ```suggestion
For more advanced use cases, the REST API and web UI also accept an optional `graphql_query_id` parameter to specify a custom GraphQL query to use when rendering the configuration. If a `graphql_query_id` is not provided, the default query configured in the Device's Golden Config settings will be used.
```

> Starting in Nautobot v2.4.2, this REST API endpoint also accepts a `branch` query parameter to specify the branch of the Jinja2 templates Git repository to use when rendering the configuration.

Author: ```suggestion
Starting in Nautobot v2.4.2, the REST API and UI also accept an optional `branch` parameter to specify the branch of the Jinja2 templates Git repository to use when rendering the configuration.
```
